### ISSUE TITLE
Missing Terrain Tiles in Tiberian Dawn Map Editor

### ISSUE THREAD

**Issue by RadicalEdward2** [OpenRA/OpenRA#12687]
Most of the beach/coast, and rive tiles are missing from the Tiberian Dawn map editor. 
I don't know the exact name of the assets but, here's a list of missing tiles:
- upward facing waterfall (river)
- left/right facing waterfalls (river)
- coast/beach corners (more specifically the " \ " shaped one)
- upward facing river to ocean outlet
- left and right river to ocean outlets
- most (if not all) of the traversable small rock debris

I'm relatively new to both the github and OpenRA community but, I just wanted to help out by addressing which map tiles are missing because their absence severely limits what types of maps can be made.

I was going to make remakes of the maps from C&C: Renegade (Field, Islands, Under, etc) for Tiberian Dawn but, had to put them all on hold because I can't find any of the tiles listed above.

**Comment by Mailaender** [comment]
Can you make screenshots of the tiles you can't find?

**Comment by pchote** [comment]
Some (I suspect all) of these are missing because they simply don't exist.  The TD tilesets have many less tiles than RA because they were not designed with user-created maps in mind.

**Comment by AoAGeneral** [comment]
I don't remember those kinds of waterfalls but I will inspect the maps in greater detail later. I know off the top of my head that two maps did use snow debri and trees on temperate maps that existed in two main GDI missions which were never used again. (Not even in Covert Ops). 

GDI Mission 9: http://pnmedia.gamespy.com/planetcnc.gamespy.com/games/td/maps/scg09ea.png

GDI Mission 8 (1st): http://pnmedia.gamespy.com/planetcnc.gamespy.com/games/td/maps/scg08ea.png

**Comment by pchote** [comment]
@AoAGeneral: that's the `WINTER` tileset.

**Comment by AoAGeneral** [comment]
Aye. Just stating it doesn't exist in ORA editor.

**Comment by pchote** [comment]
?

![screen shot 2017-02-04 at 11 57 32](https://cloud.githubusercontent.com/assets/167819/22618313/365c8a66-ead1-11e6-95d2-c7d1251a6a20.png)

**Comment by AoAGeneral** [comment]
Winter doesn't have temperate grass tiles.

**Comment by reaperrr** [comment]
> Winter doesn't have temperate grass tiles.

Uh, it does. Winter != Snow.

**Comment by AoAGeneral** [comment]
Aye quite right. Good signs for me to stay off

**Comment by reaperrr** [comment]
Pretty sure at least these are missing due to lack of TD art/tmp files:

- upward facing waterfall (river)
- coast/beach corners (more specifically the " \ " shaped one)
- upward facing river to ocean outlet
- left and right river to ocean outlets
- most (if not all) of the traversable small rock debris
- left/right facing waterfalls (river)

The only ones I'm not sure about are
- left and right shallow river crossings for jungle

since Jungle is a custom tileset.

**Comment by RadicalEdward2** [comment]
I would understand that some assets for Snow weren't made because they were never used in-game considering Westwood didn't introduce an editor until Red Alert.

At the same time, considering OpenRA is a fan-made modernized port, making duplicates of the missing shore tiles and river tiles shouldn't be too difficult considering a lot of the assets are more-or-less identical from TD to RA (aside from extremely subtle difference in grass or dirt color).

Having not experimented with the OpenRA iteration of the RA editor, it might have the missing tiles. 
Also JUNGLE isn't too different from TEMPERAT so recreating some of the missing beach tiles shouldn't be to difficult. 
I wouldn't know myself since I'm more familiar with the visual side as opposed to the programing side but, I do know that TEMPERAT (either in TD or RA) has all the beach and/or river tiles that JUNGLE and SNOW are missing.

**Comment by RadicalEdward2** [comment]
Okay. So I made a visual for the missing beach tiles if this helps.
[img]http://i.imgur.com/bX3yURl.png[/img]

**Comment by MustaphaTR** [comment]
I remember Nooze making the missing tiles for TD, maybe we can use those if they really didn't exist in original.

**Comment by RadicalEdward2** [comment]
While I do understand that tiles don't exist in some places, couldn't some be exported over from RA to TD? I know RA has all of the missing beach tiles for every theme except the Winter/Snow one and the Jungle theme but, the Jungle only makes sense because it was a custom-made theme (that strangely was in cnc.net RA but, not in OpenRA RA). 
While naval combat isn't a thing in TD, the palette is definitely suffering from the lack of tile pieces to even make a complete coastline.

**Comment by Nyerguds** [comment]
I know this is old, but it's still open so I figure I could give my 5 cents...

Snow does not originally exist in TD. The Snow theater available in my patch is simply converted from RA, and stripped to remove the many tiles that don't exist in the first game (such as water cliffs, river disappearing into a cliff, etc). So there's no use in debating missing assets in it; it all comes from RA1 anyway.

References to a Jungle theater exist in C&C1's internals, but no official files ever existed for it, so that one I guess is added purely by OpenRA?

As for the coast lines, yeah, there are several classic mapping tricks to compensate that. I went over a few of them here:
https://forums.cncnet.org/topic/8458-terrain-tricks-northern-island-corners/

**Comment by GraionDilach** [comment]
>References to a Jungle theater exist in C&C1's internals, but no official files ever existed for it, so that one I guess is added purely by OpenRA?

Yes, https://forums.cncnet.org/topic/2058-jungle-tileset/page/8/ and related (even you were there).

**Comment by Nyerguds** [comment]
Ah, wow, that was a while ago, yes.

**Comment by GraionDilach** [comment]
Yeah, I had to trace back from #5330 since the references were pointing towards the cnc-comm domain.
